### PR TITLE
Follow up to #2185

### PR DESCRIPTION
**Pull request by jrudolph**
I mostly split this up in three separate commits. Regarding the HttpEntity changes, I cleaned up a few things around where the limitable stages were added before.

### CODE REVIEW

**Comment by jrudolph** [comment]
@raboof, here's the diff to your original PR:

https://github.com/akka/akka-http/compare/wipDecodeRequestWithMaxSize...jrudolph:for-diff-2185

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 1]
```diff
@@ -576,29 +566,31 @@ object HttpEntity {
   object LastChunk extends LastChunk("", Nil)
 
   /**
-   * Turns the given source into one that respects the `withSizeLimit` calls when used as a parameter
-   * to entity constructors.
+   * Deprecated: no-op, not explicitly needed any more.
    */
+  @deprecated("Not needed explicitly any more. ", "10.1.4")
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: 10.1.4 already released so should be 10.1.5, right?

Comment (raboof) [reply akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: CONTRIBUTING.md says "src/main/mima-filters/<last-released-version>.backwards.excludes" but this always confuses me so I'm not entirely confident...

Comment (raboof) [reply akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: In contrast to MiMa :laughing:

**Comment by jrudolph** [comment]
PLS BUILD

**Pull request merged**